erpnext/stock/report/stock_balance/stock_balance.py

```diff
@@ -36,8 +37,14 @@ def execute(filters=None):
 			qty_dict.bal_val, qty_dict.val_rate,
 			item_reorder_level,
 			item_reorder_qty,
-			company
-		])
+			company,
+
+		]
+		report_data += [get_variant_attr_val(item,i) for i in variants_attributes]
```

Comment:
Doing a query for each row is not good. Instead create a dict for all item attribute values outside for loop and then use that dict to get attribute value.

```
{
  item1: {
    'color': 'blue',
    'size': 'XL'
  },
  item2: {
    'size': 'L',
    'pattern': 'Stripe'
  }
}
```

For example, check https://github.com/frappe/erpnext/blob/develop/erpnext/accounts/report/sales_register/sales_register.py#L28